Makefile

```diff
@@ -93,3 +93,37 @@ validate-hlsl: $(SNAPSHOTS_BASE_OUT)/hlsl/*.hlsl
 		done; \
 		echo "======================"; \
 	done
+
+validate-hlsl-fxc: SHELL:=/bin/bash # required because config files uses arrays
+validate-hlsl-fxc: $(SNAPSHOTS_BASE_OUT)/hlsl/*.hlsl
+	@set -e && for file in $^ ; do \
+		FXC_PARAMS="-Zi -Od"; \
+		echo "Validating" $${file#"$(SNAPSHOTS_BASE_OUT)/"}; \
+		config="$$(dirname $${file})/$$(basename $${file}).config"; \
+		. $${config}; \
+		for (( i=0; i<$${#vertex[@]}; i++ )); do \
+			name=`echo $${vertex[i]} | cut -d \: -f 1`; \
+			profile=`echo $${vertex[i]} | cut -d \: -f 2`; \
+			sm=`echo $${profile} | cut -d \_ -f 2`; \
+			if (( sm < 6 )); then \
```

Comment:
Is it necessary to add shader model version check?

Reply:
FXC doesn't support any SM >= 6. When we add f16 support which requires SM6.2 the validation would fail but shouldn't.